Tasks.md

```diff
@@ -1,94 +1,301 @@
+    - then access that index of the `moles` array
+    - return that element
+- inside `peep()`, call the function you created and assign it to the `mole` variable
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+// returns a random element from the holes array
+function randomMole() {
+    const index = Math.floor(Math.random() * moles.length)
+    const mole = moles[index]
+    return mole
+}
+
+async function peep() {
+    const mole = randomMole()
+    // ...
+}
+```
+</details>
+
+
+### 8) SCORE
+- in the beginning of the file, add a variable for the score
+    - question: do we use `const` or `let`?
+- set it to `0` when the game starts, before the call to `peep()`
+- show it when the game starts (update the HTML `scoreBoard` element, use the `textContent` property)
+- increase it by 1 when you wack a mole (i.e. in the `whack()` function)
+    - hint: first, write the expression for "your score plus 1 point". Then assign this expression to the score-keeping variable
+- show the new score immediately, like we did in `startGame()`
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+let score
+
+// ...
+
+function startGame() {
+    score = 0
+    scoreBoard.textContent = score
+    // ...
+}
+
+function whack(event) {
+    // ...
+    score = score + 1
+    scoreBoard.textContent = score
+}
+```
+</details>
+
+
+### 9) TIMEUP
+- add a variable for whether the time is up or not
+    - which type?
+- inside `peep()`, add the condition that you don't do anything if the time is up!
+- when the game starts, set the variable to `false`
+- after 15 seconds, finish the game by setting the variable to `true`
+    - hint: create a function `finish()` which sets the variable (to what?)
+    - before calling `finish()`, use the `sleep(time)` function
+    - before `sleep` write `await` and make the `startGame()` function `async`
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+let timeUp
+
+// ...
+
+function finish() {
+    timeUp = true
+}
+
+async function startGame() {
+    timeUp = false
+    // ...
+    await sleep(15000)
+    finish()
+}
+
+async function peep() {
+    if (!timeUp) {
+        // ...
+    }
+}
+```
+</details>
+
+
+### 10) RESTART GAME
+- the `setInterval` function actually returns an object, let's call it `clock` and assign it to a variable
+- after finishing the game, call `clearInterval()` passing the `clock` as parameter
+- add a log to the `peep()` function to see if it worked!
+- now we're able to restart the game. Inside the `finish()` function, re-enable the start button
+
+<details>
+<summary>Feeling lost? Expand to get a hint!</summary>
+
+```javascript
+function finish() {
+    // ...
+    startButton.disabled = false
+}
+
+async function startGame() {
+    // ...
+    const clock = setInterval(peep, 1400)
+    // ...
+    clearInterval(clock)
```

Comment:
Probably here we need to show that `const clock` needs to come after `peep`, and `clearInterval` needs to come after `finish`. Probably something like:
```javascript
    // ... right after peep() was called
    const clock = setInterval(peep, 1400)
    // ... after finish was called
    clearInterval(clock)
```